.github/workflows/build.yaml

```diff
@@ -0,0 +1,54 @@
+name: build-refdocs
+
+on:
+  push:
+    branches:
+      - master
+    tags:
+  pull_request:
+
+jobs:
+  build:
+    name: Build reference docs
+
+    strategy:
+      matrix:
+        go-version: [1.15.x]
+        platform: [ubuntu-latest]
+
+    runs-on: ${{ matrix.platform }}
+
+    env:
+      BUILD_PATH: ${{ github.workspace }}/tmp
+
+    steps:
+
+      - name: Set up Go ${{ matrix.go-version }}
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ matrix.go-version }}
+        id: go
+
+      - name: Check out reference-docs code
+        uses: actions/checkout@v2
+
+      - name: Set release version
+        id: version_value
+        run: |
+          mkdir -p ${{env.BUILD_PATH}}
+          cd ${{env.BUILD_PATH}} && wget https://raw.githubusercontent.com/kubernetes/website/master/config.toml
+          echo 'KUBE_VERSION<<EOF' >> $GITHUB_ENV
+          awk '/^version/ {print}' ${{env.BUILD_PATH}}/config.toml | cut -c 12-16 >> $GITHUB_ENV
+          echo 'EOF' >> $GITHUB_ENV
+
+      - name: Build references
+        run: |
+          echo "Building comp and gen-resourcesdocs"
+          echo "$GITHUB_WORKSPACE"
+          mkdir -p ${{env.BUILD_PATH}}/src/k8s.io
+          echo "${{env.BUILD_PATH}}"
+          wget https://github.com/kubernetes/kubernetes/archive/${{env.KUBE_VERSION}}.0.tar.gz -O ${{env.BUILD_PATH}}/src/k8s.io/kubernetes-src.tar.gz
+          pushd ${{env.BUILD_PATH}}/src/k8s.io && tar xzf kubernetes-src.tar.gz && mv kubernetes-${{env.KUBE_VERSION}}.0 kubernetes && popd
+          pushd ${{env.BUILD_PATH}}/src/k8s.io/kubernetes && make generated_files && popd
+          make comp
+          make genresources
```

Comment:
After the build, what should happen? Is the aim of the Action to test that a build works / do we want to save the output?

Reply:
Right now, travis captures the build output and displays success or failure? 
I don't know if there is similar functionality available with GitHub Actions.

Reply:
We'd need a place to put the output artifact(s) I think. I don't believe GitHub offers that bit.

Reply:
I think it is okay to build the source and report success/failure (do I need to add a code coverage action).
You could imagine saving the generated Markdown files (as artifacts) and adding a new step to run `hugo serve`.
My main focus at this point is to swap out the .travis script and build the source files.
Not sure if the G.A. builds will be any faster.